app.js

```diff
@@ -45,24 +43,50 @@ arrayOfFiles.forEach(function(file) {
   discoveryDocs.push(path.join('./data/breaches/', file));
 });
 
-// let discoveryParams; // discoveryParams will be set after Discovery is validated and setup.
+// setup Discovery by finding/creating environment and finding/creating collection
+var discoverySetupParams = { default_name: DEFAULT_COLLECTION_NAME };
 const discoverySetup = new WatsonDiscoverySetup(discovery);
-const partialDocs = discoveryDocs.slice(0, 15);
-const discoverySetupParams = { default_name: DEFAULT_NAME, documents: partialDocs };
 discoverySetup.setupDiscovery(discoverySetupParams, (err, data) => {
   if (err) {
     handleSetupError(err);
   } else {
-    console.log('Discovery is ready!' + data);
-    // discoveryParams = data;
+    console.log('Discovery is ready!');
+
+    // now load data into discovery service collection
+    var collectionParams = data;
+
+    // set collection creds - at this point the collectionParams
+    // will point to the actual credentials, whether the user
+    // entered them in .env for an existing collection, or if
+    // we had to create them from scratch.
+    environment_id = collectionParams.environment_id;
```

Comment:
I think environment_id and collection_id should already be in discoverySetupParams. By collecting disco param stuff there, I was able to just forward it on and re-use it or copy into more specific params w/ Object.assign.  So it might be another thing that app.js doesn't need to see.